source/Calamari/Kubernetes/Authentication/AzureKubernetesServicesAuth.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes.Integration;
+
+namespace Calamari.Kubernetes.Authentication
+{
+    public class AzureKubernetesServicesAuth
+    {
+        readonly AzureCli azureCli;
+        readonly Kubectl kubectlCli;
+        readonly IVariables deploymentVariables;
+
+        /// <summary>
+        /// This class is responsible for configuring the kubectl auth against an AKS cluster for an Octopus Deployment
+        /// </summary>
```

Comment:
I really like this one-liner summary document comment. Sometimes I feel the class/method names are not the best place to express the intention (and just like comments, names can get out of date as well). We should probably use more of these.